GVRf/Framework/jni/oculus/activity_jni.cpp

```diff
@@ -75,150 +77,292 @@ GVRActivity::~GVRActivity() {
+    jmethodID mid = UiJni->GetMethodID(activityClass, name, signature);
+    if (!mid) {
+        FAIL( "couldn't get %s", name);
     }
     return mid;
 }
 
-
-jclass GVRActivity::GetGlobalClassReference( const char * className ) const
-{
+jclass GVRActivity::GetGlobalClassReference(const char * className) const {
     jclass lc = UiJni->FindClass(className);
-    if ( lc == 0 )
-    {
-        FAIL( "FindClass( %s ) failed", className );
+    if (lc == 0) {
+        FAIL( "FindClass( %s ) failed", className);
     }
     // Turn it into a global ref, so we can safely use it in the VR thread
-    jclass gc = (jclass)UiJni->NewGlobalRef( lc );
+    jclass gc = (jclass) UiJni->NewGlobalRef(lc);
 
-    UiJni->DeleteLocalRef( lc );
+    UiJni->DeleteLocalRef(lc);
 
     return gc;
 }
 
 void GVRActivity::Configure( OVR::ovrSettings & settings )
 {
-    settings.EyeBufferParms.multisamples = 2;
-    // leave the rest as default for now.
-    // TODO: take values specified in xml and set them here.
+    //General settings.
+    JNIEnv *env = app->GetVrJni();
+    jobject vrSettings = env->CallObjectMethod(javaObject,
+            getAppSettingsMethodId);
+    jint framebufferPixelsWide = env->GetIntField(vrSettings,
+            env->GetFieldID(vrAppSettingsClass, "framebufferPixelsWide", "I"));
+    if (framebufferPixelsWide == -1) {
+        app->GetVrJni()->SetIntField(vrSettings,
+                env->GetFieldID(vrAppSettingsClass, "framebufferPixelsWide",
+                        "I"), settings.FramebufferPixelsWide);
+    } else {
+        settings.FramebufferPixelsWide = framebufferPixelsWide;
+    }
+    jint framebufferPixelsHigh = env->GetIntField(vrSettings,
+            env->GetFieldID(vrAppSettingsClass, "framebufferPixelsHigh", "I"));
+    if (framebufferPixelsHigh == -1) {
+        env->SetIntField(vrSettings,
+                env->GetFieldID(vrAppSettingsClass, "framebufferPixelsHigh",
+                        "I"), settings.FramebufferPixelsHigh);
+    } else {
+        settings.FramebufferPixelsHigh = framebufferPixelsHigh;
+    }
+    settings.ShowLoadingIcon = env->GetBooleanField(vrSettings,
+            env->GetFieldID(vrAppSettingsClass, "showLoadingIcon", "Z"));
+    settings.UseSrgbFramebuffer = env->GetBooleanField(vrSettings,
+            env->GetFieldID(vrAppSettingsClass, "useSrgbFramebuffer", "Z"));
+    settings.UseProtectedFramebuffer = env->GetBooleanField(vrSettings,
+            env->GetFieldID(vrAppSettingsClass, "useProtectedFramebuffer",
+                    "Z"));
+
+
+
+    //Settings for EyeBufferParms.
+    jobject eyeParmsSettings = env->GetObjectField(vrSettings,
+            env->GetFieldID(vrAppSettingsClass, "eyeBufferParms",
+                    "Lorg/gearvrf/utility/VrAppSettings$EyeBufferParms;"));
+    jclass eyeParmsClass = env->GetObjectClass(eyeParmsSettings);
+    settings.EyeBufferParms.multisamples = env->GetIntField(eyeParmsSettings, env->GetFieldID(eyeParmsClass, "multiSamples", "I"));
```

Comment:
so i put a log after this line to print out the value for multisamples.  no matter what i put in the xml file, this still ends up being 2.